x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolverTests.java

```diff
@@ -0,0 +1,512 @@
+        return new TestEnrichPolicyResolver(cluster, new HashMap<>(), new HashMap<>(), new HashMap<>());
+    }
+
+    class TestEnrichPolicyResolver extends EnrichPolicyResolver {
+        final String cluster;
+        final Map<String, EnrichPolicy> policies;
+        final Map<String, String> aliases;
+        final Map<String, Map<String, String>> mappings;
+
+        TestEnrichPolicyResolver(
+            String cluster,
+            Map<String, EnrichPolicy> policies,
+            Map<String, String> aliases,
+            Map<String, Map<String, String>> mappings
+        ) {
+            super(
+                mockClusterService(policies),
+                transports.get(cluster),
+                new IndexResolver(new FieldCapsClient(threadPool, aliases, mappings), cluster, EsqlDataTypeRegistry.INSTANCE, Set::of)
+            );
+            this.policies = policies;
+            this.cluster = cluster;
+            this.aliases = aliases;
+            this.mappings = mappings;
+        }
+
+        EnrichResolution resolvePolicies(Collection<String> clusters, Collection<UnresolvedPolicy> unresolvedPolicies) {
+            PlainActionFuture<EnrichResolution> future = new PlainActionFuture<>();
+            if (randomBoolean()) {
+                unresolvedPolicies = new ArrayList<>(unresolvedPolicies);
+                for (Enrich.Mode mode : Enrich.Mode.values()) {
+                    for (String policy : List.of("hosts", "address", "author")) {
+                        if (randomBoolean()) {
+                            unresolvedPolicies.add(new UnresolvedPolicy(policy, mode));
+                        }
+                    }
+                }
+                if (randomBoolean()) {
+                    unresolvedPolicies.add(new UnresolvedPolicy("legacy-policy-1", randomFrom(Enrich.Mode.values())));
+                }
+            }
+            super.resolvePolicies(clusters, unresolvedPolicies, future);
+            return future.actionGet(30, TimeUnit.SECONDS);
+        }
+
+        @Override
+        protected Transport.Connection getRemoteConnection(String remoteCluster) {
+            assertThat("Must only called on the local cluster", cluster, equalTo(LOCAL_CLUSTER_GROUP_KEY));
+            return transports.get("").getConnection(transports.get(remoteCluster).getLocalDiscoNode());
+        }
+
+        static ClusterService mockClusterService(Map<String, EnrichPolicy> policies) {
+            ClusterService clusterService = mock(ClusterService.class);
+            EnrichMetadata enrichMetadata = new EnrichMetadata(policies);
+            ClusterState state = ClusterState.builder(new ClusterName("test"))
+                .metadata(Metadata.builder().customs(Map.of(EnrichMetadata.TYPE, enrichMetadata)))
+                .build();
+            when(clusterService.state()).thenReturn(state);
+            return clusterService;
+        }
+    }
+
+    static class FieldCapsClient extends FilterClient {
```

Comment:
Interesting. There were many times in the past when I needed a mocked up fieldcaps client.